remoteworkitem/remoteworkitem.go

```diff
@@ -34,25 +41,54 @@ const (
 // WorkItemKeyMaps relate remote attribute keys to internal representation
 var WorkItemKeyMaps = map[string]WorkItemMap{
 	ProviderGithub: WorkItemMap{
-		AttributeExpression(GithubTitle):       models.SystemTitle,
-		AttributeExpression(GithubDescription): models.SystemDescription,
-		AttributeExpression(GithubState):       models.SystemState,
-		AttributeExpression(GithubID):          models.SystemRemoteItemID,
-		AttributeExpression(GithubCreator):     models.SystemCreator,
-		AttributeExpression(GithubAssignee):    models.SystemAssignee,
+		AttributeMapper{AttributeExpression(GithubTitle), StringConverter{}}:       models.SystemTitle,
+		AttributeMapper{AttributeExpression(GithubDescription), StringConverter{}}: models.SystemDescription,
+		AttributeMapper{AttributeExpression(GithubState), GithubStateConverter{}}:  models.SystemState,
+		AttributeMapper{AttributeExpression(GithubID), StringConverter{}}:          models.SystemRemoteItemID,
+		AttributeMapper{AttributeExpression(GithubCreator), StringConverter{}}:     models.SystemCreator,
+		AttributeMapper{AttributeExpression(GithubAssignee), StringConverter{}}:    models.SystemAssignee,
 	},
 	ProviderJira: WorkItemMap{
-		AttributeExpression(JiraTitle):    models.SystemTitle,
-		AttributeExpression(JiraBody):     models.SystemDescription,
-		AttributeExpression(JiraState):    models.SystemState,
-		AttributeExpression(JiraID):       models.SystemRemoteItemID,
-		AttributeExpression(JiraCreator):  models.SystemCreator,
-		AttributeExpression(JiraAssignee): models.SystemAssignee,
+		AttributeMapper{AttributeExpression(JiraTitle), StringConverter{}}:      models.SystemTitle,
+		AttributeMapper{AttributeExpression(JiraBody), StringConverter{}}:       models.SystemDescription,
+		AttributeMapper{AttributeExpression(JiraState), GithubStateConverter{}}: models.SystemState,
+		AttributeMapper{AttributeExpression(JiraID), StringConverter{}}:         models.SystemRemoteItemID,
+		AttributeMapper{AttributeExpression(JiraCreator), StringConverter{}}:    models.SystemCreator,
+		AttributeMapper{AttributeExpression(JiraAssignee), StringConverter{}}:   models.SystemAssignee,
 	},
 }
 
+type AttributeConverter interface {
+	Convert(interface{}) (interface{}, error)
+}
+
+type StateConverter interface{}
+
+type StringConverter struct{}
+
+type GithubStateConverter struct{}
+
+// Convert method map the external tracker item to ALM WorkItem
+func (sc StringConverter) Convert(value interface{}) (interface{}, error) {
+	return value, nil
+}
+
+func (ghc GithubStateConverter) Convert(value interface{}) (interface{}, error) {
+	if value.(string) == "closed" {
```

Comment:
Wondering if the Converter should have access to the full Tracker Item as well. That would allow us to map GH states based on a combination of state and labels.